src/pytorch_lightning/core/module.py

```diff
@@ -1737,7 +1746,7 @@ def tbptt_split_batch(self, batch, split_size):
                 if isinstance(x, Tensor):
                     split_x = x[:, t : t + split_size]
                 elif isinstance(x, collections.Sequence):
-                    split_x = [None] * len(x)
+                    split_x = [None] * len(x)  # type: ignore[assignment]
```

Comment:
would this be solved if we just wrote it as a list comprehension (the loop below)?

Reply:
I used

```python
split_x = [x[batch_idx][t : t + split_size] for batch_idx in range(len(x))]
```

and got

```sh
src/pytorch_lightning/core/module.py:1750: error: Incompatible types in assignment (expression has type "List[Any]", variable has type "Tensor")  [assignment]
```